### PR TITLE
Fix dom_import_simplexml() return type

### PR DESCRIPTION
**Pull request by cmb69**
See <https://github.com/php/php-src/pull/16489>.

### CODE REVIEW

File: reference/dom/functions/dom-import-simplexml.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
  <refsect1 role="description">
   &reftitle.description;
   <methodsynopsis>
-   <type>DOMElement</type><methodname>dom_import_simplexml</methodname>
+   <type class="union"><type>DomAttr</type><type>DOMElement</type></type><methodname>dom_import_simplexml</methodname>
```
Comment (nielsdos) [inline reference/dom/functions/dom-import-simplexml.xml:14]:
```suggestion
   <type class="union"><type>DOMAttr</type><type>DOMElement</type></type><methodname>dom_import_simplexml</methodname>
```

**Review by nielsdos — CHANGES_REQUESTED** [review]

**Comment by nielsdos** [comment]
The documentation of this function is also a bit whack...

Description:
> This function takes the node node of class [SimpleXML](https://www.php.net/manual/en/ref.simplexml.php) and makes it into a [DOMElement](https://www.php.net/manual/en/class.domelement.php) node. This new object can then be used as a native [DOMElement](https://www.php.net/manual/en/class.domelement.php) node.

This is inaccurate now


Return values:
> The [DOMElement](https://www.php.net/manual/en/class.domelement.php) node added.

What ???

File: reference/dom/functions/dom-import-simplexml.xml (hunk 1 above)
Comment (cmb69) [reply reference/dom/functions/dom-import-simplexml.xml]: Oh, that was wishful thinking, Thank you for checking!

**Comment by cmb69** [comment]
> What ??? What does added even mean here?

Indeed, makes no sense, so updated. (a bit sad, though, since it would have had its 20th birthday in a couple of weeks ;)

Before updating/improving/fixing the description, could you please have a look at the implementation of dom_import_simplexml_common():

https://github.com/php/php-src/blob/42e179ef9dc7f5b8cf6f1135c6be8e7ffe4aa5cb/ext/dom/php_dom.c#L566-L570

Is it really okay to cast an arbitrary object to a `php_libxml_node_object`?

**Comment by nielsdos** [comment]
Note that `php_libxml_import_node` will fail if the object is not actually an xml object.
I suppose it would be cleaner to check failure of `php_libxml_import_node` first and only then do the cast, but I don't think the current ordering hurts.

**Comment by cmb69** [comment]
Ah, I've overlooked `php_libxml_import_node`!

I've pushed a further commit regarding the function and the parameter descriptions. However, it seems to me that not only `SimpleXMLElement`s can be imported, but rather arbitrary objects which have been registered with libxml. Might be worth to document that at some time.

**Comment by nielsdos** [comment]
Indeed it's broader than simplexml. However, in practice only DOM & simplexml register handlers to be able to do it.
Also a small nit on wording: you used "convert", but it's not actually converting in the sense that it's a clone; they're working on the same xml nodes so it's more like creating a "view". i.e. changes you make via dom are gonna be visible in simplexml and vice versa.

**Comment by cmb69** [comment]
Thanks for the clarification! I tried to improve the wording, and added a respective example.

**Review by nielsdos — APPROVED** [review]

**Comment by nielsdos** [comment]
Thanks a lot!

**Pull request merged**